lib/config/config-file.js

```diff
@@ -538,10 +538,8 @@ function loadFromDisk(resolvedPath, configContext) {
             }
         }
 
-        const ruleMap = configContext.linterContext.getRules();
-
-        // validate the configuration before continuing
-        validator.validate(config, resolvedPath.configFullName, ruleMap.get.bind(ruleMap), configContext.linterContext.environments);
+        validator.validateConfigSchema(config, resolvedPath.configFullName);
+        validator.validateEnvironment(config.env, resolvedPath.configFullName, configContext.linterContext.environments);
```

Comment:
I make the change since there are some mocked configs have some undefined rules.
rule options could be validated when the rules being loaded(in `linter.js`).